### PR TITLE
Cleanup and reorganise code in sidebar

### LINKED ISSUE #2908: Simplify logic, DOM and class names on sidebar.js
After the 2021-02-05, do the following improvements on `sidebar.js`:

* remove the feature flag `disableShadowSidebar`
* review and simplify the logic
* review the DOM organisation and class names

### PR DESCRIPTION
**Pull request by esanzgar**
- Remove `disableShadowSidebar` flag
- Simplify code structure
- Remove iframe from externalContainer on `destroy`.

Closes #2908

### CODE REVIEW

**Comment by robertknight** [comment]
I rebased this on master.

File: src/annotator/sidebar.js [hunk 1]
```diff
@@ -183,8 +174,11 @@ export default class Sidebar extends Guest {
   destroy() {
     this._unregisterEvents();
     this._hammerManager?.destroy();
-    this.frame?.remove();
-    this.hypothesisSidebar?.remove();
+    if (this.hypothesisSidebar) {
+      this.hypothesisSidebar.remove();
```
Comment (robertknight) [inline src/annotator/sidebar.js:178]: To check my understanding - this is because if `this.hypothesisSidebar` exists, we assume it contains `this.iframe` and so that no longer needs to be removed?

File: src/annotator/sidebar.js [hunk 2]
```diff
@@ -63,54 +63,50 @@ export default class Sidebar extends Guest {
...
+    this.iframe = createSidebarIframe(config);
+
+    const {
+      externalContainerSelector,
+      theme,
+      openSidebar,
+      annotations,
+      query,
+      group,
+    } = config;
```
Comment (robertknight) [inline src/annotator/sidebar.js:75]: My subjective view here is that this use of destructuring isn't a win for readability, because some of these variables are used only much later in the function and it is now less obvious that they are just values from the config.

File: src/annotator/sidebar.js [hunk 3]
```diff
@@ -63,54 +63,50 @@ export default class Sidebar extends Guest {
...
+    if (externalContainerSelector) {
+      this.externalFrame =
+        /** @type {HTMLElement} */
+        (document.querySelector(externalContainerSelector)) ?? element;
+      this.externalFrame.appendChild(this.iframe);
     } else {
-      frame = document.createElement('div');
-      frame.style.display = 'none';
-      frame.className = 'annotator-frame';
+      this.iframeContainer = document.createElement('div');
```
Comment (robertknight) [inline src/annotator/sidebar.js:83]: This is definitely a better name than `frame`.

File: src/annotator/test/sidebar-test.js [hunk 4]
```diff
@@ -760,6 +762,21 @@ describe('Sidebar', () => {
...
       });
+
+      it('removes the iframe from the container when destroyed', () => {
+        sidebar.show();
+        assert.exists(sidebar.iframe.parentElement);
+        sidebar.destroy();
+        assert.notExists(sidebar.iframe.parentElement);
+      });
+
+      it('ignores pan events', () => {
```
Comment (robertknight) [inline src/annotator/test/sidebar-test.js:773]: This test seems unrelated to the other changes in this PR. Can you clarify why it was added?

**Review by robertknight — COMMENTED** [review]
There was one test added here which seems unrelated to the other changes in this PR. I also had a minor quibble with the use of destructuring in one function. Otherwise the changes look good.

File: src/annotator/sidebar.js (hunk 1 above)
Comment (esanzgar) [reply src/annotator/sidebar.js:178]: Correct. `this.iframe` element is a children of the hypothesisSidebar's shadow DOM.

A new addition is that now we remove `this.iframe` from the `externalContainer` on `destroy`. That's was something overlooked previously. That's the reason for the extra test.

File: src/annotator/sidebar.js (hunk 2 above)
Comment (esanzgar) [reply src/annotator/sidebar.js:75]: I agree, I will leave them as they were.

File: src/annotator/sidebar.js (hunk 3 above)
Comment (esanzgar) [reply src/annotator/sidebar.js:83]: Thanks. The first time I took a look at the class I found confusing the `this.frame` naming.

File: src/annotator/test/sidebar-test.js (hunk 4 above)
Comment (esanzgar) [reply src/annotator/test/sidebar-test.js:773]: A new addition is that now we remove `this.iframe` from the `externalContainer` on `destroy`. That's was something overlooked previously. That's the reason for the extra test.

**Comment by esanzgar** [comment]
@robertknight I reverted the restructuring of the config variable on a second commit.

I will rebase together the two commits before merging.

**Comment by robertknight** [comment]
Rebased on master.

**Review by robertknight — APPROVED** [review]
LGTM. I'll get this merged and deployed out as Eduardo is away today.
